src/Particle.js

```diff
@@ -672,39 +847,42 @@ class Particle {
-	 * @param  {String} [options.product]  Event for this product ID or slug
-	 * @param  {String} options.auth      Access Token
+	 * @param {Object} options             Options for this API call
+	 * @param {String} options.name        Event name
+	 * @param {String} options.data        Event data
+	 * @param {Boolean} options.isPrivate  Should the event be publicly available?
+	 * @param {String} [options.product]   Event for this product ID or slug
+	 * @param {String} options.auth        Access Token
+	 * @param {Object} [options.headers]   Key/Value pairs like `{ 'X-FOO': 'foo', X-BAR: 'bar' }` to send as headers.
+	 * @param {Object} [options.context]   Request context
 	 * @returns {Promise} A promise
 	 */
-	publishEvent({ name, data, isPrivate, product, auth, context }){
+	publishEvent({ name, data, isPrivate, product, auth, headers, context }){
 		const uri = product ? `/v1/products/${product}/events` : '/v1/devices/events';
 		const postData = { name, data, private: isPrivate };
-		return this.post(uri, postData, auth, context);
+		return this.post({ uri, auth, headers, data: postData, context });
 	}
 
 	/**
 	 * Create a webhook
-	 * @param  {Object} options Options for this API call
-	 * @param  {String} options.deviceId           Trigger webhook only for this device ID or Name
-	 * @param  {String} options.name               Webhook name
-	 * @param  {String} options.url                URL the webhook should hit
-	 * @param  {String} [options.requestType=POST]        HTTP method to use
-	 * @param  {Object} [options.headers]            Additional headers to add to the webhook
-	 * @param  {Object} [options.json]               JSON data
-	 * @param  {Object} [options.query]              Query string data
-	 * @param  {String} [options.body]               Custom webhook request body
-	 * @param  {Object} [options.responseTemplate]   Webhook response template
-	 * @param  {Object} [options.responseTopic]      Webhook response topic
-	 * @param  {Boolean} [options.rejectUnauthorized] Reject invalid HTTPS certificates
-	 * @params {Boolean} [options.noDefaults]        Don't include default event data in the webhook request
-	 * @param  {Object} [options.webhookAuth]        HTTP Basic Auth information
-	 * @param  {Object} [options.form]               Form data
-	 * @param  {String} [options.product]          Webhook for this product ID or slug
-	 * @param  {String} options.auth               Access Token
+	 * @param {Object} options                        Options for this API call
+	 * @param {String} options.deviceId               Trigger webhook only for this device ID or Name
+	 * @param {String} options.name                   Webhook name
+	 * @param {String} options.url                    URL the webhook should hit
+	 * @param {String} [options.requestType=POST]     HTTP method to use
+	 * @param {Object} [options.headers]              Additional headers to add to the webhook
```

Comment:
🤦‍♂ always a gotcha. `.createWebhook()` has always accepted a `headers` object but in this case it's used to configure the _webhook's_ headers.

it feels super-foot-gun-y to leave this as-is but i wasn't sure what others thought. i'm inclined to tweak the option names such that `headers` always refers to the request headers and `<webhook headers>` uses a different field name 🤷‍♂

Reply:
@monkbroc this is what i came up with 👉 https://github.com/particle-iot/particle-api-js/pull/115/commits/4dc6551df1230b61875ae0d59bb9b833a54cd539